pandora-client-web/src/components/header/leaveButton.tsx

```diff
@@ -0,0 +1,139 @@
+
+export function LeaveButton() {
+	const [show, setShow] = useState(false);
+	const closeDialog = useCallback(() => setShow(false), []);
+
+	return (
+		<leaveButtonContext.Provider value={ closeDialog }>
+			<HeaderButton icon={ logoutIcon } iconAlt='Leave' onClick={ () => setShow(true) } title='Leave' />
+			{
+				show ? <DialogLeave /> : null
+			}
+		</leaveButtonContext.Provider>
+	);
+}
+
+function DialogLeave(): ReactElement {
+	const closeDialog = useContext(leaveButtonContext);
+
+	return (
+		<ModalDialog>
+			<Column className='LeaveDialog'>
+				<ChatRoomLeave />
+				<CharacterLeave />
+				<AccountLeave />
+				<Button onClick={ closeDialog }>Cancel</Button>
+			</Column>
+		</ModalDialog>
+	);
+}
+
+function ChatRoomLeave(): ReactElement | null {
+	const player = usePlayer();
+	const room = useChatRoomInfo();
+
+	if (!player || !room) {
+		return null;
+	}
+
+	return (
+		<CharRoomLeaveInner player={ player } room={ room } />
+	);
+}
+
+function CharRoomLeaveInner({ player, room }: { player: PlayerCharacter; room: IChatRoomFullInfo; }): ReactElement | null {
+	const directoryConnector = useDirectoryConnector();
+	const playerState = usePlayerState();
+	const canLeave = useCharacterRestrictionsManager(playerState, player, (manager) => (manager.isInSafemode() || !manager.getEffects().blockRoomLeave));
+	const [leaveButtonRef, setLeaveButtonRef] = useState<HTMLButtonElement | null>(null);
+
+	const onRoomLeave = useCallback(() => {
+		directoryConnector.awaitResponse('chatRoomLeave', EMPTY)
+			.then((result) => {
+				if (result.result !== 'ok') {
+					toast(`Failed to leave room:\n${result.result}`, TOAST_OPTIONS_ERROR);
+				}
+			})
+			.catch((err) => {
+				GetLogger('LeaveRoom').warning('Error during room leave', err);
+				toast(`Error during room creation:\n${err instanceof Error ? err.message : String(err)}`, TOAST_OPTIONS_ERROR);
+			});
+	}, [directoryConnector]);
+
+	return (
+		<fieldset>
+			<legend>Chat Room</legend>
+			<span>Name: { room.name }</span>
+			<span>Id: { room.id }</span>
+			{
+				!canLeave ? (
+					<HoverElement parent={ leaveButtonRef } className='action-warning'>
+						An item is preventing you from leaving the room.
+					</HoverElement>
+				) : null
+			}
+			<Button onClick={ onRoomLeave } className='fadeDisabled' disabled={ !canLeave } ref={ setLeaveButtonRef }>
+				Leave room
+			</Button>
+		</fieldset>
+	);
+}
+
+function CharacterLeave(): ReactElement | null {
+	const directoryConnector = useDirectoryConnector();
+	const connectionInfo = useShardConnectionInfo();
+	const characterData = usePlayerData();
+	const characterName = (characterData && !characterData.inCreation) ? characterData.name : null;
+
+	const onClick = useCallback((e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+		e.preventDefault();
+		e.stopPropagation();
+		directoryConnector.sendMessage('disconnectCharacter', EMPTY);
+	}, [directoryConnector]);
+
+	if (!connectionInfo) return null;
```

Comment:
I think instead of not showing it, it is much better to show the area, but instead of button show that you don't have a character selected.